rasa/core/brokers/pika.py

```diff
@@ -223,7 +225,11 @@ async def _set_up_exchange(
     async def _bind_queue(
         queue_name: Text, channel: aio_pika.RobustChannel, exchange: aio_pika.Exchange
     ) -> None:
-        queue = await channel.declare_queue(queue_name, durable=True)
+        try:
+            queue = await channel.declare_queue(queue_name, durable=True)
+        except ChannelNotFoundEntity as e:
```

Comment:
How about handling the exception [here](https://github.com/RasaHQ/rasa/blob/5a9b83bacda02f0a293ef40c7e394afdc82de868/rasa/core/brokers/broker.py#L33) together with the others? Would having to duplicate the logic and makes sure we're handling exceptions upon the broker creation consistently. I'd also throw `ConnectionException` (it's a subclass of `RasaException`) instead of `RasaException` directly as it's more specific.

Reply:
I think for consistency it makes sense to have it where you suggest. But in general it is better not to leak specific implementation exceptions into the base class.